### PR TITLE
Set session and indexers' data files rel to AppDataPath

### PR DESCRIPTION
**Pull request by dubeg**
The setting AppDataPath now defaults to being relative to the working directory.
The session svc's PROVIDER_CONFIG now defaults to AppDataPath/data/sessions.
The issue indexer's IssuePath now defaults to AppDataPath/indexers/issues.bleves.

This change targets [#2159](https://github.com/go-gitea/gitea/issues/2159).

### CODE REVIEW

**Comment by cez81** [comment]
Also fixes #1378

File: models/models.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func LoadConfigs() {
 	DbCfg.Timeout = sec.Key("SQLITE_TIMEOUT").MustInt(500)
 
 	sec = setting.Cfg.Section("indexer")
-	setting.Indexer.IssuePath = sec.Key("ISSUE_INDEXER_PATH").MustString("indexers/issues.bleve")
+	setting.Indexer.IssuePath = sec.Key("ISSUE_INDEXER_PATH").MustString(path.Join(setting.AppDataPath, "indexers/issues.bleve"))
```
Comment (lafriks) [inline models/models.go]: It needs to be checked if path is absolute (see https://github.com/go-gitea/gitea/blob/master/modules/setting/setting.go#L842)

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by dubeg** [comment]
I am not used to the pull request process. I amended my current commit to include the feedback. Is this the correct method?

I added a new setting variable containing the working directory named `AppWorkPath`. Maybe I should've named it `AppWorkDir`?

**Comment by lafriks** [comment]
Now it is called differently in multiple places so either dir or path but do not mix (i would vote for path). You can add commits to your branch and they will show up here

**Comment by dubeg** [comment]
Ok I changed to **workPath** the fn & vars that were named **workDir**. Anything else?

**Review by lafriks — APPROVED** [review]

**Comment by dubeg** [comment]
I just added a minor change to the path ProviderConfig. I forgot to remove the "data" in "data/sessions" when joining with AppDataPath. 

By the way, I noticed we use `path.Join` as well as `filepath.Join`  almost as often. I checked their definition and they're a bit different. What's the logic here? Should we use one or the other, or does it not matter?

**Comment by lafriks** [comment]
@dubeg Please fix formatting issues (see drone build that is failing). `path.Join` should be used for URL's, `filepath.Join` for file path :)

**Comment by lunny** [comment]
@dubeg need rebase

**Comment by lunny** [comment]
need rebase.

**Comment by dubeg** [comment]
Sorry I've been terribly busy and also lazy since August. I'll try to fix the PR this week. As for filepath to path, I didn't know the purpose of each of them until Lafriks commented on them. I'll review what I did on that matter.

**Comment by lunny** [comment]
@dubeg Or could you give the permission to let maintainers to help you to push to this PR.

**Comment by dubeg** [comment]
@lunny How can I do that? 
Is it related to this checkbox?
editmaintainers

**Comment by lunny** [comment]
@dubeg yes

**Comment by dubeg** [comment]
@lunny  Well it's been checked for a while. If you want to finish to modify the PR, go ahead. Otherwise I'll work on it this weekend.

**Comment by lunny** [comment]
@dubeg rebased and LGTM

File: modules/setting/setting.go [hunk 2]
```diff
@@ -1188,7 +1187,7 @@ func NewXORMLogService(disableConsole bool) {
 			if err = os.MkdirAll(path.Dir(logPath), os.ModePerm); err != nil {
 				panic(err.Error())
 			}
-			logPath = filepath.Join(filepath.Dir(logPath), "xorm.log")
+			logPath = path.Join(filepath.Dir(logPath), "xorm.log")
```
[the remark this answers is not in the document]
Comment (lafriks) [reply modules/setting/setting.go]: @lunny please change to `filepath` methods for files

**Comment by lafriks** [comment]
LGTM

**Pull request merged**